### PR TITLE
Create operation - return created instance id in a location option.

### PR DESCRIPTION
**Pull request by anttiylitokola**
* Fix error IOTCLT-433

### CODE REVIEW

**Comment by anttiylitokola** [comment]
@teetak01, @AnttiKauppila

File: source/m2mobjectinstance.cpp [hunk 1]
```diff
@@ -656,16 +656,44 @@ sn_coap_hdr_s* M2MObjectInstance::handle_post_request(nsdl_s *nsdl,
...
+                            snprintf(obj_inst_id, 10, "%d",M2MBase::instance_id());
+
+                            obj_name += M2MBase::name();
+                            obj_name += "/";
+                            obj_name += obj_inst_id;
+                            obj_name += "/";
+                            obj_name += resource_id;
+
+                            coap_response->options_list_ptr->location_path_len = obj_name.length();
+                            coap_response->options_list_ptr->location_path_ptr = (uint8_t*)malloc(coap_response->options_list_ptr->location_path_len);
```
Comment (AnttiKauppila) [inline source/m2mobjectinstance.cpp]: length should be checked for != 0 (undef behaviour in malloc)
Also if length > 0 newly created pointer must be NULL checked

File: source/m2mobjectinstance.cpp [hunk 2]
```diff
@@ -656,16 +656,44 @@ sn_coap_hdr_s* M2MObjectInstance::handle_post_request(nsdl_s *nsdl,
...
                                                                 M2MTLVDeserializer::Post);
+
+                    uint16_t instance_id = deserializer->instance_id(received_coap_header->payload_ptr);
                     switch(error) {
                         case M2MTLVDeserializer::None:
                             if(observation_handler) {
                                 observation_handler->value_updated(this);
-                            }
+                            }                            
+                            coap_response->options_list_ptr = (sn_coap_options_list_s*)malloc(sizeof(sn_coap_options_list_s));
```
Comment (AnttiKauppila) [inline source/m2mobjectinstance.cpp]: Newly created pointer must be NULL checked

File: source/m2mobject.cpp [hunk 3]
```diff
@@ -482,7 +485,24 @@ sn_coap_hdr_s* M2MObject::handle_post_request(nsdl_s *nsdl,
...
+                                        memset(coap_response->options_list_ptr, 0, sizeof(sn_coap_options_list_s));
+
+                                        obj_name = M2MBase::name();
+                                        obj_name += String("/");
+                                        obj_inst_id = (char*)malloc(10);
+                                        snprintf(obj_inst_id, 10,"%d",instance_id);
+                                        obj_name += obj_inst_id;
+
+                                        coap_response->options_list_ptr->location_path_len = obj_name.length();
+                                        coap_response->options_list_ptr->location_path_ptr = (uint8_t*)malloc(coap_response->options_list_ptr->location_path_len);
```
Comment (AnttiKauppila) [inline source/m2mobject.cpp]: length should be checked for != 0 (undef behaviour in malloc)
Also if length > 0 newly created pointer must be NULL checked

File: source/m2mobject.cpp [hunk 4]
```diff
@@ -482,7 +485,24 @@ sn_coap_hdr_s* M2MObject::handle_post_request(nsdl_s *nsdl,
                                         if(observation_handler) {
                                             observation_handler->value_updated(this);
                                         }
+                                        coap_response->options_list_ptr = (sn_coap_options_list_s*)malloc(sizeof(sn_coap_options_list_s));
```
Comment (AnttiKauppila) [inline source/m2mobject.cpp]: Newly created pointer must be NULL checked

**Pull request merged**